Makefile.docker

```diff
@@ -37,7 +39,7 @@ check-build-image-manifest-up-to-date: update-build-image-tag ## Update build im
 
 .PHONY: commit-new-image-tag
 commit-new-image-tag: update-build-image-tag ## Update build image manifest and commits the changes
-	git commit --quiet $(image_manifest_file) $(build_image_tag_file) Dockerfile --message 'Update build image manifest and tag file'
+	git commit --quiet $(image_manifest_file) $(build_image_tag_file) $(update_generated_workflow_file) Dockerfile --message 'Update build image manifest and tag file'
```

Comment:
maybe this can do something like
```
git stash
update-build-image-tag
git add -u
git commit -m "..."
git stash pop
```
?

Reply:
that still leaves open files currently added but not committed as stash won't hide those (to my understanding). Is this to avoid scenario where we accidentally commit changes to those 3 files that were not intended?

Reply:
first and foremost it's to avoid having to explicitly list the files. the `stash` part is to avoid accidentally committing other files. `update-build-image-tag` shouldn't create new files

Reply:
makes sense, sounds like a worthwhile change. I'll add it to this PR